src/app/components/Account/AccountLinkWithAddressSwitch.tsx

```diff
@@ -0,0 +1,39 @@
+import { FunctionComponent, useEffect, useState } from 'react'
+import { AddressSwitchOption } from '../AddressSwitch'
+import { AccountLink, AccountLinkProps } from './AccountLink'
+import { getOasisAddress, isValidEthAddress, isValidOasisAddress } from '../../utils/helpers'
+
+interface WithAddressSwitchProps {
+  addressSwitchOption: AddressSwitchOption
+}
+
+interface WrappedComponentProps extends AccountLinkProps, WithAddressSwitchProps {}
+
+const withAddressSwitch =
+  (Component: FunctionComponent<AccountLinkProps>): FunctionComponent<WrappedComponentProps> =>
+  (props: WrappedComponentProps) => {
+    const { addressSwitchOption, address, ...restProps } = props
+
+    const [convertedAddress, setConvertedAddress] = useState<string>('')
+
+    useEffect(() => {
+      const resolveAddresses = async () => {
+        if (addressSwitchOption === AddressSwitchOption.Oasis && isValidOasisAddress(address)) {
+          setConvertedAddress(address)
+        } else if (addressSwitchOption === AddressSwitchOption.ETH && isValidEthAddress(address)) {
+          setConvertedAddress(address)
+        } else if (addressSwitchOption === AddressSwitchOption.Oasis && isValidEthAddress(address)) {
+          const oasisAddress = await getOasisAddress(address)
+          setConvertedAddress(oasisAddress)
+        } else {
+          setConvertedAddress(address)
```

Comment:
this looks weird. in three cases we do nothing, but call ` setConvertedAddress(address)`. can we simplify this?